src/main/java/mate/academy/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+package mate.academy.bookshop.service.impl;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.order.OrderItemResponseDto;
+import mate.academy.bookshop.dto.order.OrderRequestDto;
+import mate.academy.bookshop.dto.order.OrderResponseDto;
+import mate.academy.bookshop.dto.order.UpdateOrderDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.exception.InvalidStatusException;
+import mate.academy.bookshop.mapper.OrderItemMapper;
+import mate.academy.bookshop.mapper.OrderMapper;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.model.order.Order;
+import mate.academy.bookshop.model.order.OrderItem;
+import mate.academy.bookshop.model.order.Status;
+import mate.academy.bookshop.repository.CartItemRepository;
+import mate.academy.bookshop.repository.OrderItemRepository;
+import mate.academy.bookshop.repository.OrderRepository;
+import mate.academy.bookshop.repository.ShoppingCartRepository;
+import mate.academy.bookshop.service.OrderService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(OrderRequestDto requestDto, Long userId) {
+        if (!shoppingCartRepository.existsByUserId(userId)) {
+            throw new EntityNotFoundException("No shopping cart for user id " + userId);
+        }
+
+        ShoppingCart userShoppingCart = shoppingCartRepository.findByUserId(userId);
```

Comment:
Better to make one request to db instead of 2, you can shoppingCartRepository.findByUserId(userId).orElse(...)